SmartDeviceLink-iOS/SmartDeviceLink/SDLLocalization.m

```diff
@@ -0,0 +1,198 @@
+// SDLLocalization.m
+//
+
+
+#import "SDLLocalization.h"
+
+/** Private definition extending setter for the properties. */
+@interface SDLLocalization ()
+
+@property (readwrite) NSString *language;
+
+@property (readwrite) NSString *region;
+
+@property (readwrite) NSBundle *defaultBundle;
+
+@property (readwrite) NSLocale *defaultLocale;
+
+@property (readwrite) NSBundle *fallbackBundle;
+
+@property (readwrite) NSLocale *fallbackLocale;
+
+@end
+
+@implementation SDLLocalization
+
+- (NSBundle *)bundle {
+    if ([self defaultBundle]) {
+        return [self defaultBundle];
+    } else if ([self fallbackBundle]) {
+        return [self fallbackBundle];
+    } else {
+        return nil;
+    }
+}
+
+- (NSLocale *)locale {
+    if ([self defaultLocale]) {
+        return [self defaultLocale];
+    } else if ([self fallbackLocale]) {
+        return [self fallbackLocale];
+    } else {
+        return nil;
+    }
+}
+
++ (instancetype)defaultLocalization {
+    static SDLLocalization *object = nil;
+    static dispatch_once_t token;
+
+    dispatch_once(&token, ^{
+        // get the locale identifier
+        NSString *localeIdentifier = [[NSLocale currentLocale] localeIdentifier];
```

Comment:
You probably don't want to be using the locale to identify language. If I'm in Japan with my phone, I'm all of a sudden going to start seeing Japenese on my car radio, which is obviously undesirable. ([See this Stack answer](http://stackoverflow.com/questions/1522210/nslocale-currentlocale-always-returns-en-us-not-users-current-language) as proof that this is the incorrect call) Instead, there are two ways to approach this. ([Using this Stack Answer](http://stackoverflow.com/questions/3910244/getting-current-device-language-in-ios))

1. `NSString * language = [NSLocale preferredLanguages].firstObject;` This will give you the two letter code for the currently selected device language.
2. `[[NSBundle mainBundle] preferredLocalizations].firstObject;` This will give you the localization of the current app language.

I'm guessing that the first is what we want, since they could theoretically have localizations for SDL that they don't have for their main app, but I'd be willing to hear others' feedback on that.